search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java

```diff
@@ -184,13 +188,63 @@ public SearchQueryResult<ProcessDefinitionEntity> searchProcessDefinitions(
   @Override
   public SearchQueryResult<ProcessInstanceEntity> searchProcessInstances(
       final ProcessInstanceQuery filter) {
+    if (!filter.filter().incidentErrorHashCodes().isEmpty()) {
+      return searchProcessInstancesByIncidentErrorHash(filter);
+    }
+
     return getSearchExecutor()
         .search(
             filter,
             io.camunda.webapps.schema.entities.operate.listview.ProcessInstanceForListViewEntity
                 .class);
   }
 
+  private SearchQueryResult<ProcessInstanceEntity> searchProcessInstancesByIncidentErrorHash(
+      final ProcessInstanceQuery filter) {
+
+    final var originalFilter = filter.filter();
+
+    final var incidentFilter =
+        FilterBuilders.incident(
+            f ->
+                f.incidentErrorHashCodes(originalFilter.incidentErrorHashCodes())
+                    .states(IncidentState.ACTIVE));
+
+    final var incidentResult = searchIncidents(IncidentQuery.of(f -> f.filter(incidentFilter)));
+    searchIncidents(IncidentQuery.of(f -> f.filter(incidentFilter)));
+
+    if (incidentResult.items().isEmpty()) {
+      return new SearchQueryResult.Builder<ProcessInstanceEntity>().build();
+    }
+
+    final var errorMessageOperations =
+        incidentResult.items().stream()
+            .map(incident -> Operation.eq(incident.errorMessage()))
+            .toList();
```

Comment:
❓Wouldn't it be better to map these to `processInstanceKey`? And then add those to the modified query.